lib/spec/openapi/index.js

```diff
@@ -20,9 +20,12 @@ module.exports = function (opts, cache, routes, Ref, done) {
     const openapiObject = prepareOpenapiObject(defOpts, done)
 
     ref = Ref()
+    const definitions = Object.entries(ref.definitions().definitions)
+      .filter(([, schema]) => defOpts.hideSchemas.findIndex(hideSchema => hideSchema === schema.$id) < 0)
+      .reduce((defs, [$id, schema]) => ({ ...defs, [$id]: schema }), {})
```

Comment:
Can you change this code into a single `for loop`?
It repeatedly cloning the `defs`.

Reply:
Sure, I chaged it.